On Wed, Apr 12, 2017 at 11:47 AM, Jim Fulton <jim@jimfulton.info> wrote:

>
>
> On Wed, Apr 12, 2017 at 10:50 AM, Jens W. Klein <notifications@github.com>
> wrote:
>
>> Ok, this shows that the original commit already comes with the problem.
>> As it was green at some point it must be something introduced later.
>>
>> @plone/testing-team <https://github.com/orgs/plone/teams/testing-team>
>> and @jimfulton <https://github.com/jimfulton> any idea why it fails so
>> badly here?
>>
> I'm confused.  The tests were passing and now aren't.
>
> I just ran the non-selenium tests (/bin/alltsests) on use-current-zodb-3
> on my mac and they all passed.
>
> Total time elapsed: 1585.193 seconds
>
> Grand total: 90 packages, 0 failures
>
> Are the failures selenium tests?
>
> I'll try on linux via a docker container.
>


I ran the tests in a linux container.  I'm actually unsure how to interpret
results from bin/alltests.

The summary:


Packages with test failures:

Failing tests in group plone_app_testing

Total time elapsed: 1474.068 seconds

Grand total: 90 packages, 1 failures

But if I scroll up:

Error in test testNormalizeLatin1
(Products.CMFPlone.tests.testUnicodeSplitter.TestCaseNormalizer)
Traceback (most recent call last):
  File "/usr/lib/python2.7/unittest/case.py", line 329, in run
    testMethod()
  File
"/home/jim/p/plone/buildout.coredev/src/Products.CMFPlone/Products/CMFPlone/tests/testUnicodeSplitter.py",
line 127, in testNormalizeLatin1
    saved = _setlocale(*LATIN1)
  File
"/home/jim/p/plone/buildout.coredev/src/Products.CMFPlone/Products/CMFPlone/tests/testUnicodeSplitter.py",
line 28, in _setlocale
    str(LATIN1))
Error: ('Unsupported locale. These tests need at least one of the following
locales available on your system', "('en_US.ISO-8859-1',
'en_US.ISO8859-15', 'en_GB.ISO8859-15', 'de_DE@euro', 'fr_FR@euro',
'nl_NL@euro')")

    1074/1143 (94.0%)

Error in test testProcessLatin1
(Products.CMFPlone.tests.testUnicodeSplitter.TestSplitter)
Traceback (most recent call last):
  File "/usr/lib/python2.7/unittest/case.py", line 329, in run
    testMethod()
  File
"/home/jim/p/plone/buildout.coredev/src/Products.CMFPlone/Products/CMFPlone/tests/testUnicodeSplitter.py",
line 89, in testProcessLatin1
    saved = _setlocale(*LATIN1)
  File
"/home/jim/p/plone/buildout.coredev/src/Products.CMFPlone/Products/CMFPlone/tests/testUnicodeSplitter.py",
line 28, in _setlocale
    str(LATIN1))
Error: ('Unsupported locale. These tests need at least one of the following
locales available on your system', "('en_US.ISO-8859-1',
'en_US.ISO8859-15', 'en_GB.ISO8859-15', 'de_DE@euro', 'fr_FR@euro',
'nl_NL@euro')")

Gaaa, 2 errors that don't seem to show up in the summary. I'm guessing I
can fix these by installing some local in my container.

Oh, more of these above.

I also note a traceback with:

    browser.open(portalURL+'/@@manage-dashboard?_authenticator=' +
createToken())

Is this trying to run a selenium test?

I don't see a failure in the output.

--- a/versions.cfg
+++ b/versions.cfg
@@ -40,6 +40,11 @@ zope.pagetemplate = 3.6.3
 # Support for more complex catalog queries
 Products.ZCatalog = 3.0.2
 Zope2 = 2.13.26
+ZEO = 5.1.0
+zodbpickle = 0.6.0
+futures = 3.0.5
+python-jose = 1.3.2
+trollius = 2.1
 
 #############
 # Build Tools

--- a/versions.cfg
+++ b/versions.cfg
@@ -25,8 +25,11 @@ BTrees = 4.4.1
 # Overrides until ztk is updated
 coverage = 4.3.4
 zdaemon = 4.2.0
-transaction = 1.7.0
+transaction = 2.1.2
 Sphinx = 1.5.2
+ZODB = 5.2.0
+ZODB3 = 3.11.0
+tempstorage = 4.0
 zope.app.locales = 3.7.5
 zope.component = 3.12.1
 zope.i18n = 4.1.0

